### PR TITLE
metrics: Cherry pick #15118 to release-6.5: add min_safe_ts, min_safe_ts_region and min_safe_ts_gap

### PR DESCRIPTION
**Pull request by ekexium**
### What is changed and how it works?

Issue Number: ref #15082 

What's Changed:

Cherry pick #15118 

```commit-message
metrics: add min_safe_ts, min_safe_ts_region and min_safe_ts_gap
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

Side effects

### Release note 

```release-note
None
```

### CODE REVIEW

File: metrics/grafana/tikv_details.json [hunk 1]
```diff
@@ -4532,7 +4532,6 @@
           },
           "yaxes": [
             {
-              "$$hashKey": "object:150",
```
Comment (MyonKeminta) [inline metrics/grafana/tikv_details.json:4535]: 🤔where are these changes come from?

**Review by MyonKeminta — COMMENTED** [review]
It seems that there are many unrelated changes in tikv_details.json. Should we avoid them or just merge ignoring them?

File: metrics/grafana/tikv_details.json (hunk 1 above)
Comment (ekexium) [reply metrics/grafana/tikv_details.json:4535]: I have no idea... ChatGPT told me this https://github.com/grafana/grafana/issues/11551 and it seems OK to just remove them

**Review by MyonKeminta — APPROVED** [review]

**Review by cfzjywxk — APPROVED** [review]

**Comment by ekexium** [comment]
@tikv/qa-release-merge PTAL

**Pull request merged**